src/_guides/language/effective-dart/usage.md

```diff
@@ -1156,6 +1233,54 @@ Future<bool> fileContainsBear(String path) async {
 }
 {% endprettify %}
 
+
+### DO test for a future when disambiguating a `FutureOr<T>` whose value type could be `Object`.
+
+Before you can do anything useful with a `FutureOr<T>`, you typically need to do
+an `is` check to see if you have a `Future<T>` or a bare `T`. If the type
+argument is some specific type like `FutureOr<int>`, it doesn't matter which
+test you use, `is int` or `is Future<int>`. Either works because those two types
+are disjoint.
+
+However, if the value type is `Object` or a type parameter that could possibly
+be instantiated with `Object` then the two types overlap. `Future<Object>`
+itself implements `Object`, so `is Object` or `is T` where `T` is some type
+parameter that could be instantiated with object returns true even when the
+object is a future. Instead, explicitly test for the `Future` case:
+
+{:.good-style}
+<?code-excerpt "misc/lib/effective_dart/usage_good.dart (test-future-or)"?>
+{% prettify dart %}
+Future<T> logValue<T>(FutureOr<T> value) async {
+  if (value is Future<T>) {
+    var result = await value;
+    print(result);
+    return result;
+  } else {
+    print(value);
+    return value as T;
+  }
+}
+{% endprettify %}
+
+{:.bad-style}
+<?code-excerpt "misc/lib/effective_dart/usage_bad.dart (test-future-or)"?>
+{% prettify dart %}
+Future<T> logValue<T>(FutureOr<T> value) async {
+  if (value is T) {
+    print(value);
+    return value;
+  } else {
+    var result = await value;
+    print(result);
+    return result;
+  }
+}
+{% endprettify %}
```

Comment:
How about eliding the if-then-else bodies? We don't need them to make the point that testing against `value is T` is a bad idea.